utilities/post_migration.py

```
@@ -708,6 +708,42 @@ def check_network(source_vm: dict[str, Any], destination_vm: dict[str, Any], net
+        if not source_nic_name:
+            continue
+
+        source_mac = source_nic.get("macAddress", "").lower()
+
+        for dest_nic in dest_nics:
+            if dest_nic.get("macAddress", "").lower() == source_mac:
+                dest_nic_name = dest_nic.get("guest_interface_name", "")
+                assert dest_nic_name == source_nic_name, (
+                    f"NIC name mismatch: source {source_nic_name} != destination {dest_nic_name} (MAC {source_mac})"
+                )
+                LOGGER.info(f"NIC name preserved: {source_nic_name} (MAC {source_mac})")
```

Ruff (0.15.14)

[warning] Logging statement uses f-string

(G004)